File: inverter_hil/python/inverterhilgui/format_pin_state.py
import math
from .gui_theme import guiTheme


def formatPinState(state):
    theme = guiTheme()
    result = {"text": theme["text"]["noData"], "color": theme["color"]["secondaryText"], "known": False}
    if not isinstance(state, (bool, int, float)) or not math.isfinite(float(state)) or float(state) not in (0, 1):
        return result
    result.update(text="ON" if bool(state) else "OFF",
                  color=theme["color"]["healthy"] if bool(state) else theme["color"]["disabledText"], known=True)
    return result

File: VC_HIL/inverter/+inverterhilgui/+live_telemetry/formatPinState.m
function display = formatPinState(state)
%FORMATPINSTATE Render a monitored VCU output pin as text plus color.
%
%   Plan 7.3 requires TP6-TP10 pin cards to use ON/OFF text plus color so
%   state is never conveyed by color alone. The TEXT field is therefore
%   always authoritative and is never empty; COLOR is decoration only.
%
%   DISPLAY fields: text, color, known.

theme = inverterhilgui.live_telemetry.guiTheme();
display = struct( ...
    'text', theme.text.noData, ...
    'color', theme.color.secondaryText, ...
    'known', false);

if ~(islogical(state) || isnumeric(state)) || ~isscalar(state) || ...
        ~isreal(state) || ~isfinite(double(state)) || ...
        ~(double(state) == 0 || double(state) == 1)
    return;
end

if logical(state)
    display.text = 'ON';
    display.color = theme.color.healthy;
else
    display.text = 'OFF';
    display.color = theme.color.disabledText;
end
display.known = true;
end
